docs/pages/guides/adding-delegation.mdx

```diff
@@ -0,0 +1,726 @@
+- Update the user interface.
+
+Here is the modified `packages/client/src/App.tsx`:
+
+<CollapseCode>
+
+```typescript filename="App.tsx" copy showLineNumbers {1,8-56,119-138,178-202}
+import { encodeFunctionData } from "viem";
+import { useMUD } from "./MUDContext";
+import { getBurnerPrivateKey, createBurnerAccount, transportObserver, getContract } from "@latticexyz/common";
+import { createPublicClient, createWalletClient, fallback, webSocket, http, ClientConfig, Hex } from "viem";
+import { getNetworkConfig } from "./mud/getNetworkConfig";
+import IWorldAbi from "contracts/out/IWorld.sol/IWorld.abi.json";
+
+// The ABI we need
+const ABI = [
+  {
+    inputs: [
+      {
+        internalType: "address",
+        name: "delegatee",
+        type: "address",
+      },
+      {
+        internalType: "ResourceId",
+        name: "systemId",
+        type: "bytes32",
+      },
+      {
+        internalType: "uint256",
+        name: "numCalls",
+        type: "uint256",
+      },
+    ],
+    name: "initDelegation",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [],
+    name: "newCall",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+];
+
+// Create the system name constants
+
+const name2hex = (name, length) => {
+  let result = "";
+  for (let i = 0; i < name.length; i++) result += name.charCodeAt(i).toString(16);
+
+  if (length >= name.length) return result.padEnd(length * 2, "0");
+  else return result.slice(0, length * 2);
+};
+
+const systemId = (namespace, systemName) =>
+  "0x" + name2hex("sy", 2) + name2hex(namespace, 14) + name2hex(systemName, 16);
+
+const LASTCALL_SYSTEM_ID = systemId("LastCall", "LastCallSystem");
+const SYSTEMBOUND_DELEGATION = systemId("", "systembound");
+
+// A lot of code that would normally go in mud/getNetworkConfig.ts is part of the application here,
+// because `getNetworkConfig.ts` assumes you use one wallet, and here we need several.
+const networkConfig = await getNetworkConfig();
+
+const clientOptions = {
+  chain: networkConfig.chain,
+  transport: transportObserver(fallback([webSocket(), http()])),
+  pollingInterval: 1000,
+} as const satisfies ClientConfig;
+
+const publicClient = createPublicClient({
+  ...clientOptions,
+});
+
+// Create a structure with two fields:
+//   client - a wallet client that uses a random account
+//   world - a world contract object that lets us issue newCall
+const makeWorldContract = () => {
+  const client = createWalletClient({
+    ...clientOptions,
+    account: createBurnerAccount(getBurnerPrivateKey(Math.random().toString())),
+  });
+
+  return {
+    world: getContract({
+      address: networkConfig.worldAddress as Hex,
+      abi: IWorldAbi,
+      publicClient: publicClient,
+      walletClient: client,
+    }),
+    client,
+  };
+};
+
+// Create five world contracts
+const worldContracts = [1, 2, 3, 4, 5].map((x) => makeWorldContract());
```

Comment:
wait why do we need 5 world contracts?

Reply:
Because we have five different addresses (all burner wallets). We need to have multiple addresses to show delegation.